src/sentry/options/manager.py

```diff
@@ -164,13 +172,24 @@ def delete(self, key):
 
         return self.store.delete(opt)
 
-    def register(self, key, default='', type=None, flags=DEFAULT_FLAGS,
+    def _value_is_of_type(self, type, value):
+        # TODO(dcramer): replace with basic types
+        if type in (unicode, str):
+            type = basestring
+        return isinstance(value, type)
+
+    def register(self, key, default=None, type=None, flags=DEFAULT_FLAGS,
                  ttl=DEFAULT_KEY_TTL, grace=DEFAULT_KEY_GRACE):
         assert key not in self.registry, 'Option already registered: %r' % key
         # Guess type based on the default value
         if type is None:
-            if isinstance(default, basestring):
-                type = basestring
+            # the default value would be equivilent to '' if no type / default
+            # is specified and we assume unicode for safety
+            if default is None:
+                type = unicode
```

Comment:
I know, I was suggesting a similar dance like you do with `_value_is_of_type`.

But I can go through and fix this with better types after you're done.

I'm afraid of this breaking current implementation though if it's being typed as unicode, but we attempt to set it as a str, it'll be problematic.